spec.bs

```diff
@@ -6947,7 +7073,15 @@ To <dfn>adjust bid list based on k-anonymity</dfn> given a [=list=] of [=generat
   1. Set |bidCopy|'s [=generated bid/for k-anon auction=] to false.
   1. [=Apply any component ads target to a bid=] given |bidCopy|.
   1. [=list/Append=] |bidCopy| to |bidsToScore|
-  1. If [=query generated bid k-anonymity count=] given |generatedBid| returns true:
+  1. Let |selectedReportingId| be a [=string=]-or-null that is set to null.
+  1. If |generatedBid|'s [=generated bid/selected buyer and seller reporting ID=] [=map/exists=], set |selectedReportingId| to it.
+  1. Let |igAd| be the [=interest group ad=] from |generatedBid|'s [=generated bid/interest group=]'s
+    [=interest group/ads=] whose [=interest group ad/render url=] is |generatedBid|'s
+    [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
+  1. If both of the following return true:

```

Comment:
These four lines (7081-7084) seem to be an incomplete transition. Is this correct?

```
  1. If [=query generated bid k-anonymity count=] given |generatedBid| returns true AND
     (|generatedBid|'s [=generated bid/selected buyer and seller reporting ID=] does not [=map/exists|exist=] OR
      [=query reporting ID k-anonymity count=] given |generatedBid|'s [=generated bid/interest group=], |igAd|,
      and the |generatedBid|'s [=generated bid/selected buyer and seller reporting ID=] is true), then:
    1. [=list/Append=] |generatedBid| to |bidsToScore|.
```

This formation allows you to also remove these two lines above:
```
  1. Let |selectedReportingId| be a [=string=]-or-null that is set to null.
  1. If |generatedBid|'s [=generated bid/selected buyer and seller reporting ID=] [=map/exists=], set |selectedReportingId| to it.
```

Reply:
Hmm This seemed be in a messed up state. I restructured it based on your comments. However I left creating the variables because I feel as though it makes that long if statement easier to understand. Let me know what you think.